### ISSUE TITLE
Stylesheet not working in 4.8.4

### ISSUE THREAD

**Issue by esabol** [spotbugs/spotbugs#2969]
We use `stylesheet="fancy-hist.xsl"` to generate our SpotBugs reports in our CI/CD pipeline, which builds using ant. Our `build.xml` has the following:
```xml
    <taskdef
        resource="edu/umd/cs/findbugs/anttask/tasks.properties"
        classpath="${spotbugs.home}/lib/spotbugs-ant.jar"/>
 
    <target name="spotbugs">
        <spotbugs home="${spotbugs.home}"
                  output="html"
                  outputFile="${dist.dir}/spotbugs.html"
                  stylesheet="fancy-hist.xsl" >
            <auxclasspath path="${basedir}/../libs" />
            <sourcePath path="${basedir}/src/java" />
            <class location="${build.web.dir}/WEB-INF/classes" />
        </spotbugs>
    </target>
```
This all worked for us just fine with SpotBugs 4.8.3. After upgrading to 4.8.4, however, the SpotBugs report generated by the same exact `build.xml` is clearly not using `fancy-hist.xml`. And, yes, `spotbugs.home` is defined properly in `project.properties` (which hasn't changed) and we are using the `spotbugs-ant.jar` from the 4.8.4 release.

Any idea as to what the problem is or what the solution might be? Thanks!

**Comment by gtoison** [comment]
Hello @Vogel612, I'm not sure how this was working but I was wondering if this could be a side effect of #2894

**Comment by JuditKnoll** [comment]
Maybe it can be because of https://github.com/spotbugs/spotbugs/pull/2915. Can you please share the `fancy-hist.xml`? Is there any error logged about this?

**Comment by gtoison** [comment]
I think the xsl is part of SpotBugs: https://github.com/spotbugs/spotbugs/blob/master/spotbugs/src/xsl/fancy-hist.xsl
I was wondering about the version change for the filter xsd too, that might be the problem

**Comment by esabol** [comment]
Yes, the `fancy-host.xsl` we are using is the one provided by SpotBugs, as @gtoison indicated. We haven't modified it or moved it.

**Comment by esabol** [comment]
My colleague additionally tested it with SpotBugs 4.8.2, and the stylesheet worked.

Also tested 4.8.4 with `stylesheet="${spotbugs.home}/src/xsl/fancy-hist.xsl"` instead of `stylesheet="fancy-hist.xsl"`. Still didn't work.

4.8.3 works with either of those two `stylesheet` attribute values.

**Comment by Vogel612** [comment]
> Hello @Vogel612, I'm not sure how this was working but I was wondering if this could be a side effect of #2894

The changeset you mention should only affect the filepaths, although I did touch the stylesheet assignment so I can't exclude it with any degree of certainty. Judging by the syntax of the options, the code contributed with that changeset should be working as intended and not mess with the stylesheets, assuming generating an XML report `withMessages` also works as intended...

I haven't checked the release yet, but unfortunately this seems like my changes are a more likely culprit than the filter XSD namespace changes...

**Comment by asheldon** [comment]
> assuming generating an XML report withMessages also works as intended...

When I diff SpotBugs XML output between 4.8.2 and 4.8.5, having set output to `xml:withMessages`, the shortMessage and longMessage blocks are missing from the bugInstance.

```
<     <ShortMessage>Redundant nullcheck of value known to be non-null</ShortMessage>
<     <LongMessage>Redundant nullcheck of maxuptime, which is known to be non-null in new com.example.Test(boolean, String[])</LongMessage>
```

Additionally, the `<SourceLine>` blocks, `<LocalVariable>`, etc blocks are self-closing instead of including a `<Message>`.

**Comment by esabol** [comment]
Is there a developer who can look into this? Maybe try reverting PR #2894?

**Comment by Vogel612** [comment]
@gtoison I really don't have the capacity to look into it right now. I'm happy to go for a revert, the improvement is only marginal and the regression is quite a heavy thing.

I might be able to look at it at some later point with some more test coverage around this specific element of it, but not in any reasonable time frame. I would like to see the original issue that was supposed to be fixed by that PR reopened if we revert, though :)

**Comment by asheldon** [comment]
I checked out the source for SpotBugs and on commit 6cf7b2c64073b5aab180ec64890ab5b55808956c I did `git revert 789d3979ec2b04443793e548802ab33a1d4e4aae` and rebuilt the jar locally. The locally built jar appears to work correctly in this scenario.

**Comment by hazendaz** [comment]
Closing as fixed on master, will be released in 4.9.0 as code reverted.

**Comment by esabol** [comment]
Thanks, everyone!
